launcher/agent/agent.go

```diff
@@ -90,6 +91,35 @@ func (a *agent) Attest(ctx context.Context) ([]byte, error) {
 	return resp.ClaimsToken, nil
 }
 
+func (a *agent) AttestCustom(ctx context.Context, aud string, nonce []string) ([]byte, error) {
+	challenge, err := a.client.CreateChallenge(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	principalTokens, err := a.principalFetcher(challenge.Name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get principal tokens: %w", err)
+	}
+
+	attestation, err := a.getAttestation(challenge.Nonce)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := a.client.VerifyAttestation(ctx, verifier.VerifyAttestationRequest{
+		Challenge:      challenge,
+		GcpCredentials: principalTokens,
+		Attestation:    attestation,
+		CustomAudience: aud,
+		CustomNonce:    nonce,
+	})
```

Comment:
This is heavily duplicated with Attest. We should probably have a Attest call AttestCustom with a nil `aud` and `nonce` or pull the common impl into a helper function.

Reply:
I was thinking that we just have one method and if we see there are no custom requests (or perhaps we see sts.googleapis as the aud) we can provide the token we have on file.

Reply:
Merging the two attest function here